### ISSUE TITLE
What to do with multiple entries for one video?

### ISSUE THREAD

**Issue by girst** [girst/SkipTheBS-Server#3]
When two sets of videotags exist for one video, how do we want that to be handled?

 * average them?
 * discard all but the first/last one added?
 * let the user select which one to follow? (on which criteria?)
 * something else?

**Comment by fuerbringer** [comment]
For me the purpose behind averaging start/stop times would be to 'iron out' mis-clicks where people  selected the adbreak a bit too late or too soon. So this is how I'd solve the multiple videotags problem: Only return averages of multiple submits when they are very close to eachother (kind of like merging them).

Example for a single video:
`{"start": 33, "end": 40}`
`{"start": 31, "end": 41}`
`{"start": 0, "end": 5}`
Since the first two are obviously the same adbreak, the first part in the json response will be `{"start": 32, "end": 40.5}`
while for the intro it would just return the normal `{"start": 0, "end": 5}` because it's obviously far away from being a part of the adbreak. As for the LTT example I put in the database: the ads start/stop times at the beginning and the end would not get averaged (they're not withing a couple seconds close to eachother).

So TLDR I'd average the start/stop times that are very close to eachother (1-3s) and return them as one adbreak in json. Otherwise if a start/stop time has no other start/stop times close to it, we'll just return it regularly and treat it as an adbreak. I think that would at least get rid of the problem of having multiple submits.

I hope that made at least some sense. What do you think?

**Comment by girst** [comment]
I'm sorry for not answering 'till now - i completely missed the email notification. 

That seems like a very good idea, any ideas on how to implement it? I figure, a simple `if (abs(a-b)<3)` could do the trick, albeit be it not very clean when considering edge cases and comparing multiple things.
